src/Interpreters/ProcessList.h

```diff
@@ -331,6 +331,16 @@ class ProcessList : public ProcessListBase
 
     /// List of queries
     Container processes;
+    /// List of queries for which cancellaion is in progress
+    struct CancelledProcesses
+    {
+        using Container = std::list<QueryStatus *>;
+        using Iterator = Container::iterator;
+
+        Container processes;
```

Comment:
Do we really need this list? maybe simply add `is_cancelled` flag to ProcessListEntry?

Reply:
Yep, this will be better and simpler.